doc/Reactive.md

```diff
@@ -16,42 +70,285 @@ Each [`Message<T>`](https://github.com/unoplatform/uno.extensions/blob/main/src/
 * Public _feeds_ are expected to be exposed in property getters. A caching system embedded in reactive framework will then ensure to not re-create a new _feed_ on each get.
 
-## Sources
+## Sources: How to create a _feed_
 You can build a _feed_ from different sources. Main entry point to do that is using the static class `Feed`:
 
-| Method | Usage |
-| ------ | ----- |
-| `Async` | Creates a feed from an async method. The loaded data can be refreshed using a `Signal` trigger that will re-invoke the async method. |
-| `AsyncEnumerable` | This adapts an `IAsyncEnumerable<T>` into a _feed_ |
-| `Create` | This gives you the ability to create you own _feed_ by dealing directly with _messages_. |
+Assuming this given `IWeatherService` and `ILocationService`:
+```csharp
+public interface IWeatherService
+{
+	/// <summary>
+	/// Asynchronously gets the current weather
+	/// </summary>
+	Task<WeatherInfo> GetCurrentWeather(CancellationToken ct);
+
+	/// <summary>
+	/// Asynchronously gets the details of a weather alert.
+	/// </summary>
+	Task<string> GetAlertDetails(WeatherAlert alert, CancellationToken ct);
+
+	/// <summary>
+	/// Asynchronously gets the weather forecast for the given day
+	/// </summary>
+	Task<WeatherInfo> GetWeatherForecast(DateTime date, CancellationToken ct);
+}
+
+public record WeatherInfo(double Temperature, WeatherAlert? Alert);
+
+public record WeatherAlert(Guid Id, string? Title);
+
+public interface ILocationService
+{
+	/// <summary>
+	/// Asynchronously gets the name of the current city.
+	/// </summary>
+	Task<string> GetCurrentCity(CancellationToken ct);
+
+	/// <summary>
+	/// Gets teh list of all supported cities.
+	/// </summary>
+	Task<ImmutableList<string>> GetCities(CancellationToken ct);
+}
+```
+
+### Async
+Creates a feed from an async method. The loaded data can be refreshed using a `Signal` trigger that will re-invoke the async method.
+
+```csharp
+private IWeatherService _weatherService;
+
+public IFeed<WeatherInfo> Weather => Feed.Async(async ct => await _weatherService.GetCurrentWeather(ct));
+```
+
+### AsyncEnumerable
+This adapts an `IAsyncEnumerable<T>` into a _feed_
+
+```csharp
+private IWeatherService _weatherService;
+
+public IFeed<WeatherInfo> Weather => Feed.AsyncEnumerable(() => GetWeather());
+
+private async IAsyncEnumerable<WeatherInfo> GetWeather([EnumeratorCancellation] CancellationToken ct = default)
+{
+	while (!ct.IsCancellationRequested)
+	{
+		yield return await _weatherService.GetCurrentWeather(ct);
+		await Task.Delay(TimeSpan.FromHours(1), ct);
+	}
+}
+```
+
+### Create
+This gives you the ability to create your own _feed_ by dealing directly with _messages_.
+
+> This is designed for advanced usage and should probably not be used directly in apps.
```

Comment:
```suggestion
> [!NOTE]
> This is designed for advanced usage and should probably not be used directly in apps.
```

Reply:
Also, why is this advanced? Can we link to another section which deals with these advanced scenarios?